File: Day-Four/python/CheckOut/check_out_test.py
from unittest import TestCase

import check_out_function

class TestCheckOut(TestCase):

    def test_that_get_sub_total_returns_su_total_of_the_prices(self):
        item_price_list = [5000.00, 12500.00, 3200.00, 1800.00]
        actual = check_out_function.get_sub_total(item_price_list)
        expected = 22500.0
        self.assertEqual(actual, expected)

    def test_that_get_rate_returns_ratel_in_percentage(self):
        rate = 8
        actual = check_out_function.get_rate(rate)
        expected = 0.08
        self.assertEqual(actual, expected)

    def test_that_get_discount_returns_prices_after_applying_rate(self):
        item_price_list = [5000.00, 12500.00, 3200.00, 1800.00]
        rate = 8
        actual = check_out_function.get_discount(item_price_list, rate)
        expected = 1800.0
        self.assertEqual(actual, expected)

    def test_that_get_vat_returns_prices_afterr_applying_calue_add_tax(self):
        item_price_list = [5000.00, 12500.00, 3200.00, 1800.00]
        vat = check_out_function.get_vat(item_price_list)
        expected = 3937.5
        self.assertEqual(vat, expected)

    def test_that_get_bill_total_returns_the_total_bill(self):
        item_price_list = [5000.00, 12500.00, 3200.00, 1800.00]
        rate = 8
        actual = check_out_function.get_bill_total(item_price_list, rate)
        expected = 24637.5
        self.assertEqual(actual, expected)

    def test_that_get_change_returns_difference_when_total_bill_is_greater(self):
        item_price_list = [5000.00, 12500.00, 3200.00, 1800.00]
        rate = 8
        deposit = 20000
        actual = check_out_function.get_change(deposit, item_price_list, rate)
        expected = 4637.5
        self.assertEqual(actual, expected)

    def test_that_get_change_returns_difference_when_total_bill_is_greater(self):
        item_price_list = [5000.00, 12500.00, 3200.00, 1800.00]
        rate = 8
        deposit = 30000
        actual = check_out_function.get_change(deposit, item_price_list, rate)
        expected = 5362.5
        self.assertEqual(actual, expected)

File: Day-Four/javascript/Check-Out_App/CheckOut.test.js
const { getSubTotal, getRate, getDiscount, getVat, getBillTotal, getChange } = require("./CheckOutFunction")

test("Test that get subtotal returns subtotal of the prices", ()=> {

    let itemPriceList = [5000.00, 12500.00, 3200.00, 1800.00]
    let actual = getSubTotal(itemPriceList)
    let expected = 22500.0
    
    expect(actual).toEqual(expected)

})
test("Test that get rate returns rate in percentage ", ()=> {
    
    let rate = 8;
    let actual = getRate(rate)
    let expected = 0.08
    
    expect(actual).toEqual(expected)
})

test("Test that get Disount returns total price after applying rate ", ()=> {

    let itemPriceList = [5000.00, 12500.00, 3200.00, 1800.00]
    let rate = 8
    let actual = getDiscount(itemPriceList, rate)
    let expected = 1800.0
    
    expect(actual).toEqual(expected)
})

test("Test that get vat returnd the prices after appplying value add tax ", ()=> {
    
    let itemPriceList = [5000.00, 12500.00, 3200.00, 1800.00]
    let vat = getVat(itemPriceList)
    let expected = 3937
    expect(vat).toEqual(expected)

})

test("Test that  get bill total returns total bill ", ()=> {

    let itemPriceList = [5000.00, 12500.00, 3200.00, 1800.00]  
    let rate = 8
    let actual = getBillTotal(itemPriceList, rate)
    let expected = 24637

    expect(actual).toEqual(expected)
})


test("Test that get  change returnd difference when total is greater ", ()=> {
    
    let itemPriceList = [5000.00, 12500.00, 3200.00, 1800.00]
    let rate = 8
    let deposit = 20000
    let actual = getChange(deposit, itemPriceList, rate)
    let expected = 4637

    expect(actual).toEqual(expected);
})

test("Test that get  change returnd difference when total is lesser ", ()=> {
    
    let itemPriceList = [5000.00, 12500.00, 3200.00, 1800.00]
    let rate = 8
    let deposit = 30000
    let actual = getChange(deposit, itemPriceList, rate)
    let expected = 5363

    expect(actual).toEqual(expected);
})

